src/app/pie-chart/pie-chart.component.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Component,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  OnInit,
+} from '@angular/core';
+
+@Component({
+  selector: 'app-pie-chart',
+  templateUrl: './pie-chart.component.html',
+  styleUrls: ['./pie-chart.component.scss'],
+})
+export class PieChartComponent implements OnInit, OnChanges {
+  @Input() data: any[] = [];
+  pieChartData: any[] = [];
+
+  ngOnInit(): void {
+    this.updatePieChartData();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['data'] && !changes['data'].firstChange) {
+      this.updatePieChartData();
+    }
+  }
```

Comment:
Use the same approche as the one from the ChartComponent  => refactor **PieChartComponent** to remove **OnInit** and **OnChanges** lifecycle hooks. Instead use a setter on the @Input() data property to directly trigger updateChartData method whenever 'data' changes.